graveler/graveler.go

```diff
@@ -236,7 +236,7 @@ type VersionController interface {
 	// CommitExistingMetaRange creates a commit in the branch from the given pre-existing tree.
 	// Returns ErrDirtyBranch if the branch has uncommitted changes.
 	// Returns ErrTreeNotFound if the referenced treeID doesn't exist.
```

Comment:
> This comment is really out of date:

> What does ErrDirtyBranch mean now that there is no branch?

Removed. 

> What's a "treeID" (there didn't use to be a treeID; there still is no treeID, but there didn't use to be one, either)?

Updated the comment to match the existing error.

> This interface no longer allows Graveler to advance the branch head (and, TBH, I couldn't find where the existing implementation does that). Is that intentional?

Right, branch isn't advanced here to support the API we defined. However existing functionality exists with `UpdateBranch`.

> Generally this sounds that once we have this method, either this or Commit becomes a function not a method (i.e., it could be composed of methods except maybe a line or 2 that could be added as small methods), so why force every interface implementer to implement it? It seems to have long and complex semantics, and as this PR shows it seems really focused on implementing one particular case of "import". Could we split it up maybe into a function (external to the interface) and 1, 2 or 3 added methods to do just small bits that don't already exist?

Rocks import operation is currently composed of the following:
1. ListEntries of the parent commit.
2. List the inventory (not part of the interface)
3. Write metarange with the merged iterator.
4. Commit the metarange without advancing the branch.
5. Advance the branch (if requested - plumbing will not advance).

I suggest we discuss all 5, instead of discussing of ways to split just 4.
1 & 5 are part of the existing interface, 2 isn't. So we're left with 3 & 4 which were written to suit the import flow as you mentioned. I can think of a way to merging them into one, but I'm pretty sure that breaking them further will cause more incoherent-why-do-we-need-this-oh-it's-for-import methods, not less.

Reply:
I think we're good now, especially with #1229 already appearing to re-use this interface.